### PR TITLE
[Feature] Storage volume SQL CRUD

### PR DESCRIPTION
**Pull request by TszKitLo40**
## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Problem Summary(Required):

Storage volume SQL CRUD
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 93 / 112 (83.04%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/StorageVolumeAnalyzer.java|0|3|00.00%|[94, 95, 96]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|0|16|00.00%|[900, 901, 902, 903, 908, 909, 910, 911, 916, 917, 918, 919, 925, 926, 927, 928]|
|:large_blue_circle:|com/starrocks/common/proc/StorageVolumeProcNode.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterStorageVolumeStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DescStorageVolumeStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|25|25|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateStorageVolumeStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/storagevolume/StorageVolume.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|27|27|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 109 / 112 (97.32%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/StorageVolumeAnalyzer.java|0|3|00.00%|[94, 95, 96]|
|:large_blue_circle:|com/starrocks/common/proc/StorageVolumeProcNode.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterStorageVolumeStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DescStorageVolumeStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|25|25|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateStorageVolumeStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/storagevolume/StorageVolume.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|16|16|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|27|27|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 109 / 109 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/StorageVolumeAnalyzer.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/proc/StorageVolumeProcNode.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterStorageVolumeStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DescStorageVolumeStmt.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|25|25|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateStorageVolumeStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/storagevolume/StorageVolume.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|27|27|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 108 / 110 (98.18%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|24|26|92.31%|[75, 76]|
|:large_blue_circle:|com/starrocks/sql/analyzer/StorageVolumeAnalyzer.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/proc/StorageVolumeProcNode.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterStorageVolumeStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DescStorageVolumeStmt.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateStorageVolumeStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/storagevolume/StorageVolume.java|13|13|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|27|27|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 103 / 104 (99.04%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/storagevolume/StorageVolume.java|12|13|92.31%|[101]|
|:large_blue_circle:|com/starrocks/sql/analyzer/StorageVolumeAnalyzer.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/proc/StorageVolumeProcNode.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterStorageVolumeStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/DescStorageVolumeStmt.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|22|22|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateStorageVolumeStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|25|25|100.00%|[]|

File: fe/fe-core/src/test/java/com/starrocks/storagevolume/StorageVolumeTest.java [hunk 1]
```diff
@@ -229,4 +257,186 @@ public void testHDFSInvalidCredential() {
             Assert.assertTrue(e.getMessage().contains("Storage params is not valid"));
         }
     }
+
+    @Test
+    public void testCreate() throws DdlException, AnalysisException {
+        StorageVolumeMgr storageVolumeMgr = GlobalStateMgr.getCurrentState().getStorageVolumeMgr();
+        ExceptionChecker.expectThrowsNoException(
+                () -> {
+                    String sql = "CREATE STORAGE VOLUME IF NOT EXISTS storage_volume_create type = s3 " +
```
Comment (imay) [inline fe/fe-core/src/test/java/com/starrocks/storagevolume/StorageVolumeTest.java:266]: Can you implement these cases in sql-test?

Comment (TszKitLo40) [reply fe/fe-core/src/test/java/com/starrocks/storagevolume/StorageVolumeTest.java:266]: > Can you implement these cases in sql-test?

Comment (TszKitLo40) [reply fe/fe-core/src/test/java/com/starrocks/storagevolume/StorageVolumeTest.java:266]: > Can you implement these cases in sql-test?

But if these cases are moved to sql-test, the FE unit tests coverage will fail.

Comment (kevincai) [reply fe/fe-core/src/test/java/com/starrocks/storagevolume/StorageVolumeTest.java:266]: > > Can you implement these cases in sql-test?
> 
> But if these cases are moved to sql-test, the FE unit tests coverage check will fail.

make a copy to sql-test and even add more comprehensive cases.

Comment (imay) [reply fe/fe-core/src/test/java/com/starrocks/storagevolume/StorageVolumeTest.java:266]: We will count coverage after executing sql test. So the coverage is not a solid blocker. UT should cover code that sql test is not easy to cover.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 26 / 104 (25.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AlterStorageVolumeStmt.java|0|2|00.00%|[42, 46]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateStorageVolumeStmt.java|0|2|00.00%|[62, 66]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|0|12|00.00%|[900, 901, 903, 908, 909, 911, 916, 917, 919, 925, 926, 928]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|2|25|08.00%|[368, 370, 2537, 2538, 2539, 2540, 2541, 2542, 2543, 2544, 2545, 2547, 2548, 2549, 2551, 2552, 2553, 2554, 2555, 2556, 2559, 2560, 2561]|
|:large_blue_circle:|com/starrocks/server/StorageVolumeMgr.java|2|22|09.09%|[46, 47, 48, 49, 50, 65, 66, 79, 80, 81, 82, 109, 110, 141, 145, 146, 147, 148, 149, 151]|
|:large_blue_circle:|com/starrocks/storagevolume/StorageVolume.java|2|13|15.38%|[101, 109, 135, 136, 137, 138, 139, 140, 141, 142, 143]|
|:large_blue_circle:|com/starrocks/common/proc/StorageVolumeProcNode.java|12|19|63.16%|[41, 42, 43, 44, 45, 46, 47]|
|:large_blue_circle:|com/starrocks/sql/ast/DescStorageVolumeStmt.java|3|4|75.00%|[51]|
|:large_blue_circle:|com/starrocks/sql/analyzer/StorageVolumeAnalyzer.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|2|100.00%|[]|

File: fe/fe-core/src/test/java/com/starrocks/storagevolume/StorageVolumeTest.java (hunk 1 above)
Comment (TszKitLo40) [reply fe/fe-core/src/test/java/com/starrocks/storagevolume/StorageVolumeTest.java]: > We will count coverage after executing sql test. So the coverage is not a solid blocker. UT should cover code that sql test is not easy to cover.

I have moved the tests to sql test. And there is only 25% FE UT coverage now.

**Review by sduzh — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**